lib/rules/disallow-newline-before-block-statements.js

```diff
@@ -90,15 +184,82 @@ module.exports.prototype = {
     },
 
     check: function(file, errors) {
-        file.iterateNodesByType('BlockStatement', function(node) {
+        if (this._setting === true) {
+            checkAll(file, errors);
+        } else {
+            checkSpecificTypes(file, errors, this._setting);
+        }
+    }
+};
+
+function checkAll(file, errors) {
```

Comment:
no need for this function, you can consolidate this into the checkSpecificTypes function.